channels/applicationinsights-channel-js/src/Offline.ts

```diff
@@ -1,92 +1,104 @@
-import { getWindow, getDocument, getNavigator, isUndefined, isNullOrUndefined, attachEvent } from "@microsoft/applicationinsights-core-js";
-import dynamicProto from "@microsoft/dynamicproto-js";
+import { getWindow, getDocument, getNavigator, isUndefined, isNullOrUndefined, createUniqueNamespace, mergeEvtNamespace, eventOn, eventOff } from "@microsoft/applicationinsights-core-js";
+
+export interface IOfflineListener {
+    isOnline: () => boolean;
+    isListening: () => boolean;
+    unload: () => void;
+}
+
+function _disableEvents(target: any, evtNamespace: string | string[]) {
+    eventOff(target, null, null, evtNamespace);
+}
 
 /**
- * @description Monitors browser for offline events
- * @export default - Offline: Static instance of OfflineListener
- * @class OfflineListener
+ * Create a new OfflineListener instance to monitor browser online / offline events
+ * @param parentEvtNamespace - The parent event namespace to append to any specific events for this instance
  */
-export class OfflineListener {
-    public static Offline = new OfflineListener;
-    public isListening: boolean;
-    
-    constructor() {
-        let _window = getWindow();
-        let _document = getDocument();
-        let isListening = false;
-        let _onlineStatus: boolean = true;
-
-        dynamicProto(OfflineListener, this, (_self) => {
-            try {
-                if (_window) {
-                    if (attachEvent(_window, "online", _setOnline)) {
-                        attachEvent(_window, "offline", _setOffline);
-                        isListening = true;
-                    }
-                }
-                
-                if (_document) {
-                    // Also attach to the document.body or document
-                    let target:any = _document.body || _document;
-    
-                    if (!isUndefined(target.ononline)) {
-                        target.ononline = _setOnline;
-                        target.onoffline = _setOffline;
-                        isListening = true;
-
-                    }
-                }
+export function createOfflineListener(parentEvtNamespace?: string | string[]): IOfflineListener {
```

Comment:
The event namespace is part of the changes for unload support, so the events are tagged with the current instance.